### ISSUE TITLE
[BUG] - Extension enters a lockup state and blocks all other extensions from loading

### ISSUE THREAD

**Issue by multimokia** [LuqueDaniel/vscode-language-renpy#338]
This is more of a breaking issue, I've been able to reproduce this by trying to activate jump-to-definition or hover documentation/outline population in a project without navigation data in the top level (this project cannot have that as per repo structure)

If this happens, the extension will perpetually try to load the data it needs to complete action (either docs/outline etc.) and will ramp up CPU fans. Once in this state, all other extensions that require loading will no longer finish loading either (some plugins affected are GitHub Pull Requests, GitLens, etc.) This *also* affects VSCode's git integration entirely and the source control tab will no longer update as you change files.

This issue is resolved completely by downgrading the extension to 2.1.7

**Comment by multimokia** [comment]
Also of note, it breaks other language servers from activating too (once in this state, if you attempt to open up another filetype, for example a json, none of its intellisense features work, nor code folding etc.)

This is a major breaking issue to almost any workflow

**Comment by duckdoom4** [comment]
Could you provide me with the extensions log? 
![image](https://github.com/LuqueDaniel/vscode-language-renpy/assets/60387522/2e51cd7e-2510-4c1c-8737-8f5da61de105)

**Comment by multimokia** [comment]
![image](https://github.com/LuqueDaniel/vscode-language-renpy/assets/22531674/23b6259a-4fed-4c95-8077-12f3f6fdd2ca)

Gets stuck here

**Comment by multimokia** [comment]
In this state, saving also now takes around a second or two even on small files and all language feature for other languages too no longer work, for example this settings.json file
![image](https://github.com/LuqueDaniel/vscode-language-renpy/assets/22531674/d3c8eef6-72ca-4739-8054-e89ef15e2908)

which isn't reporting the error, likewise fold regions are missing

**Comment by duckdoom4** [comment]
Alright, thanks a lot. That will help :)

How pressing is this issue? I'd prefer to work on a new feature first, but if this is blocking you from doing anything I'll swap priority

**Comment by Dingotush** [comment]
Not sure if this is the same problem, but I was doing a quick edit (a dialogue typo across multiple files) without an instance of the game running (so not ready to recompile/regenerate the outline). Once I saved the first changed file VS Code started maxing out one core. Saving the other changes I'd made wasn't happening, and any attempt to interact with Git failed. Unlike the original report my game has a conventional structure - the outline is normally present and updated.

The inability to save was a real problem. I don't think the plugin should ever be able to stop VS Code working as a text editor.

**Comment by multimokia** [comment]
No worries, it's pretty significant, running in this state is practically impossible due to the increased save times and all other plugins failing. 

For me I effectively cannot use the extension in the current release but I don't have a problem with rolling back to 2.1.7 in the meantime.

**Comment by duckdoom4** [comment]
> Not sure if this is the same problem, but I was doing a quick edit (a dialogue typo across multiple files) without an instance of the game running (so not ready to recompile/regenerate the outline). Once I saved the first changed file VS Code started maxing out one core. Saving the other changes I'd made wasn't happening, and any attempt to interact with Git failed. Unlike the original report my game has a conventional structure - the outline is normally present and updated.
> 
> The inability to save was a real problem. I don't think the plugin should ever be able to stop VS Code working as a text editor.

Yeah, that probably means the plugin is running into an infinite while loop somewhere due to some issue. Gotta figure out why first tho

**Comment by duckdoom4** [comment]
Please try v2.3.4 this issue is likely fixed in that build :)
